bridge/evm/contracts/BridgeVault.sol

```diff
@@ -59,7 +59,11 @@ contract BridgeVault is Ownable, IBridgeVault {
         recipientAddress.transfer(amount);
     }
 
-    /// @notice Enables the contract to receive ETH.
-    /// @dev This function is required to receive ETH when unwrapping WETH.
-    receive() external payable {}
+    /// @notice Wraps ass eth sent to this contract.
+    /// @dev skip if sender is wETH contract to avoid infinite loop.
+    receive() external payable {
+        if (msg.sender != address(wETH)) {
+            wETH.deposit{value: msg.value}();
+        }
```

Comment:
I don't understand this. what would happen when it's USDC?

Reply:
This function is only called when native eth is sent to the contract, in other words, no other token transfers would call this function. Working with native eth is a whole thing 🙄

Reply:
if it could only be native eth, why do we need the if check?